components/cards/ClaimAction.tsx

```diff
@@ -13,78 +15,79 @@ export type ClaimActionProps = SidebarProps & {
-  // }
-
-  // const { displayNewForSalePrice, displayNetworkFeePayment } = actionData;
+  const {
+    displayNewForSalePrice,
+    // displayNetworkFeePayment
+  } = actionData;
 
   // React.useEffect(() => {
-  //   let _transactionSubtotal = calculateWeiSubtotalField(
+  //   const _transactionSubtotal = calculateWeiSubtotalField(
   //     displayNetworkFeePayment
   //   );
 
   //   updateActionData({ transactionSubtotal: _transactionSubtotal });
   // }, [displayNetworkFeePayment]);
 
   async function _claim() {
-    // FIXME: Update for streaming payments
-    // updateActionData({ isActing: true });
-    // let baseCoord = GeoWebCoordinate.make_gw_coord(
-    //   claimBase1Coord.x,
-    //   claimBase1Coord.y
-    // );
-    // let destCoord = GeoWebCoordinate.make_gw_coord(
-    //   claimBase2Coord.x,
-    //   claimBase2Coord.y
-    // );
-    // let path = GeoWebCoordinate.make_rect_path(baseCoord, destCoord).map(
-    //   (v) => {
-    //     return BigNumber.from(v.toString(10));
-    //   }
-    // );
-    // if (path.length == 0) {
-    //   path = [BigNumber.from(0)];
-    // }
-    // const newValue = calculateWeiSubtotalField(displayNewForSalePrice);
-    // const newContributionRate = fromValueToRate(
-    //   newValue,
-    //   perSecondFeeNumerator,
-    //   perSecondFeeDenominator
-    // );
-    // const resp = await claimerContract.claim(
-    //   account,
-    //   baseCoord.toString(10),
-    //   path,
-    //   newContributionRate,
-    //   {
-    //     from: account,
-    //     value: calculateWeiSubtotalField(displayNetworkFeePayment),
-    //   }
-    // );
-    // const receipt = await resp.wait();
-    // const filter = claimerContract.filters.ParcelClaimed(null, null);
-    // const res = await claimerContract.queryFilter(
-    //   filter,
-    //   receipt.blockNumber,
-    //   receipt.blockNumber
-    // );
-    // let licenseId = res[0].args[0];
-    // return licenseId;
-    return "";
+    const baseCoord = GeoWebCoordinate.make_gw_coord(
+      claimBase1Coord.x,
+      claimBase1Coord.y
+    );
+    const destCoord = GeoWebCoordinate.make_gw_coord(
+      claimBase2Coord.x,
+      claimBase2Coord.y
+    );
+    let path = GeoWebCoordinate.make_rect_path(baseCoord, destCoord).map(
+      // FIXME: it should be 'BN' type, waiting for js-geo-web-coordinate upgrade.
+      (v: any) => {
+        return BigNumber.from(v.toString(10));
+      }
+    );
+    if (path.length == 0) {
+      path = [BigNumber.from(0)];
+    }
+    const newValue = calculateWeiSubtotalField(displayNewForSalePrice);
+    const newContributionRate = fromValueToRate(
+      newValue,
+      perSecondFeeNumerator,
+      perSecondFeeDenominator
+    );
+    const resp = await claimerContract.claim(
+      account,
+      newContributionRate,
+      ethers.utils.defaultAbiCoder.encode(
+        ["uint64", "uint256[]"],
+        [BigNumber.from(baseCoord.toString(10)), path]
+      ),
```

Comment:
new error encountered: 
```js
Error: cannot estimate gas; transaction may fail or may require manual gas limit (error=
{"code":3,"data":"...","message":"execution reverted: AccessControl: account 0x6097847fa05eb9af005224eab62784f8419cc454
 is missing role 0xf7db13299c8a9e501861f04c20f69a2444829a36a363cfad4b58864709c75560"}, 
method="estimateGas", transaction={"from":"0x6097847FA05eb9aF005224Eab62784f8419cC454","to":"0xe62D6E28597471a84daa57
392021a21696e17404","data":"...","accessList":null}, code=UNPREDICTABLE_GAS_LIMIT, 
version=providers/5.4.4)
```

I guess I am not the `Admin` according to the contract?

Reply:
More clear: 

`AccessControl: account ...
 is missing role 0xf7db13299c8a9e501861f04c20f69a2444829a36a363cfad4b58864709c75560`

Reply:
Do I need to `setClaimer` here or the contract will check `claimer` automatically?

Reply:
So the `FairLaunchClaimer` contract should not actually be called directly here, which is why you are seeing a permissions error. See the example [here](https://github.com/Geo-Web-Project/core-contracts/blob/main/docs/AuctionSuperApp.md#claim). A claim is done by creating/updating a flow. If this is the first parcel a user is claiming, the flow should be created like in that example. If there are any existing parcels, the flow should be updated and the rate should be incremented. See the [tests](https://github.com/Geo-Web-Project/core-contracts/blob/main/test/auction-super-app.ts#L208) for an example on how to update an existing flow.

Reply:
The baseCoord and path looks correct here. So you just need to change the [claimData](https://github.com/Geo-Web-Project/core-contracts/blob/main/test/auction-super-app.ts#L225) to be the value you have here.

Reply:
So, for clarity:
- `claimData` -> baseCoord, path you have here
- `actionData` -> [`forSalePrice`, `claimData`]
- `userData` -> [ Action.CLAIM, `actionData`]